src/main/scala/PSO/Particle.scala

```diff
@@ -70,42 +78,49 @@ class Particle(initVelocity: List[Double], inertia: Double, localAccel: Double,
     // Print local and neighbourhood bests to console
     case ReportRequest =>
       sender ! Report(currVel, currPos, localBest, neighbourhoodBest)
-      // println(s"Particle: lbest = $localBest, nbest = $neighbourhoodBest")
     // Handle unknown messages
     case _ =>
       throw new RuntimeException("Unknown particle message type")
   }
 
-  private def computeIteration() = {
+  def computeIteration(): Unit = {
+    iter += 1
```

Comment:
The particle doesn't know about the termination criteria, all it knows is how many times ComputeIteration has been called. This could theoretically be moved out of the particle, but it's keeping a bunch of other state so I'm not sure I see the point